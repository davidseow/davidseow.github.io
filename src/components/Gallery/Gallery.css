img {
  box-sizing: border-box;
  display: block;
  padding: 3px;
  width: 100%;
}
