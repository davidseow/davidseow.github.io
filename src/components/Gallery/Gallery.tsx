import React from "react";
import Masonry, { ResponsiveMasonry } from "react-responsive-masonry";

import "./Gallery.css";

const Gallery = (props) => {
  const { images } = props;

  return (
    <ResponsiveMasonry
      columnsCountBreakPoints={{ 350: 1, 750: 2, 900: 3 }}
      className="galleryContainer"
    >
      <Masonry className="gallery">
        {images.map((image: string, i: Number) => (
          <img key={i.toString()} src={image} alt="" />
        ))}
      </Masonry>
    </ResponsiveMasonry>
  );
};
export default Gallery;
