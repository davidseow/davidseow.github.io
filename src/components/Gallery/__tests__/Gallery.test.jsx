import React from "react";
import { render } from "@testing-library/react";
import Gallery from "../Gallery";

describe("Gallery", () => {
  it("should render", () => {
    const { asFragment } = render(<Gallery />);
    expect(asFragment()).toMatchSnapshot();
  });
});
