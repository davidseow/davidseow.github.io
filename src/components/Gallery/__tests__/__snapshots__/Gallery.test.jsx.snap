// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery should render 1`] = `
<DocumentFragment>
  <div
    style="width: 100%; height: auto; overflow-x: hidden; overflow-y: hidden; margin: auto;"
  >
    <style>
      
      .react-image-masonry-col * { width: 100%; box-sizing: border-box; }
      @keyframes fadeIn { from { opacity:0; } to { opacity:1; } }
    
    </style>
    <div
      class="react-image-masonry-col"
      style="width: 33.333333333333336%; display: flex; flex-direction: column; float: left;"
    />
    <div
      class="react-image-masonry-col"
      style="width: 33.333333333333336%; display: flex; flex-direction: column; float: left;"
    />
    <div
      class="react-image-masonry-col"
      style="width: 33.333333333333336%; display: flex; flex-direction: column; float: left;"
    />
  </div>
</DocumentFragment>
`;
